ci-operator/step-registry/cert-manager/install/cert-manager-install-commands.sh

```diff
@@ -4,97 +4,124 @@ set -e
 set -u
 set -o pipefail
 
+timestamp() {
+    date -u --rfc-3339=seconds
+}
+
 if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
     source "${SHARED_DIR}/proxy-conf.sh"
     echo "proxy: ${SHARED_DIR}/proxy-conf.sh"
 fi
 
-CATSRC=qe-app-registry
-if [[ ! "$(oc get catalogsource qe-app-registry -n openshift-marketplace -o yaml)" =~ "lastObservedState: READY" ]]; then
-    echo "The catalogsource qe-app-registry is either not existing or not ready. Will use redhat-operators to install cert-manager Operator."
-    CATSRC=redhat-operators
+# Check if the catalogsource is ready to use
+if [[ ! "$(oc get catalogsource $INSTALL_CATSRC -n openshift-marketplace -o yaml)" =~ "lastObservedState: READY" ]]; then
+    echo "Skipping as the catalogsource $INSTALL_CATSRC is either not existing or not ready."
+    exit 0
 fi
 
-oc create -f - << EOF
+# Define common variables
+SUB="openshift-cert-manager-operator"
+OPERATOR_NAMESPACE="cert-manager-operator"
+OPERAND_NAMESPACE="cert-manager"
+INTERVAL=10
+
+# Prepare the operator installation manifests
+
+echo "# Creating the Namespace, OperatorGroup and Subscription for the operator installation."
+MANIFEST=$(cat <<EOF
 apiVersion: v1
 kind: Namespace
 metadata:
-  name: cert-manager-operator
+  name: $OPERATOR_NAMESPACE
 ---
 apiVersion: operators.coreos.com/v1
 kind: OperatorGroup
 metadata:
   name: cert-manager-operator-og
-  namespace: cert-manager-operator
+  namespace: $OPERATOR_NAMESPACE
 spec:
   targetNamespaces:
-  - cert-manager-operator
+  - $OPERATOR_NAMESPACE
 ---
 apiVersion: operators.coreos.com/v1alpha1
 kind: Subscription
 metadata:
-  name: openshift-cert-manager-operator
-  namespace: cert-manager-operator
+  name: $SUB
+  namespace: $OPERATOR_NAMESPACE
 spec:
-  channel: stable-v1
-  installPlanApproval: Automatic
-  name: openshift-cert-manager-operator
-  source: "$CATSRC"
+  source: $INSTALL_CATSRC
   sourceNamespace: openshift-marketplace
+  name: openshift-cert-manager-operator
+  channel: "$INSTALL_CHANNEL"
+  installPlanApproval: Manual
```

Comment:
If $JOB_NAME does not contain operator-upgrade, use Automatic, otherwise use Manual.

Reply:
I also used to think of that. But from a step reference perspective, it should not care about if the job is install-only or install-upgrade.

I think it won't be a problem as the installplan will be auto-approved in `cert-manager-install` step.